src/components/titlePanel/createTitlePanel.ts

```diff
@@ -68,13 +79,30 @@ const createTitlePanel: TitlePanelFactory = createWidgetBase
 					};
 				}
 
+				titleProperties.role = 'heading';
+
+				if (ariaHeadingLevel) {
+					titleProperties['aria-level'] = String(ariaHeadingLevel);
+				}
+
+				if (open) {
+					// FIXME - id of content node
+					titleButtonProperties['aria-controls'] = 'content';
+				}
+
 				const children: DNode[] = [
-					v('div', titleProperties, [ title ])
+					v('div', titleProperties, [
+						v('div', titleButtonProperties, [ title ])
```

Comment:
Nitpicking, sorry! I personally think a `button` element is preferable to a `div`, but if you prefer the div, it'll also need `tabindex="0"` set (a focus manager wouldn't set initial `tabindex` values). You'd also need to make sure keyboard events (enter and space) trigger the content opening -- I'm not completely sure they work by default on a fake button.